torch/_inductor/ir.py

```diff
@@ -5186,6 +5186,11 @@ def codegen_kwargs(self, skip_out=False):  # type: ignore[no-untyped-def]
 
     def codegen_size_asserts(self, wrapper) -> None:  # type: ignore[no-untyped-def]
         if config.size_asserts and not V.graph.cpp_wrapper:
+            # if there is some symbol in size or stride, we couldn't generate asserts
```

Comment:
Why can't we generate asserts for symbolic shapes?

Reply:
I'm not very sure about my approach here is right.

The test case I meet is `python test/inductor/test_torchinductor_codegen_dynamic_shapes.py DynamicShapesCodegenCpuTests.test_custom_op_unbacked_symints_dynamic_shapes_cpu`

It may generate some code like:
```
def call(args):
    arg0_1, arg1_1, arg2_1, arg3_1 = args
    args.clear()
    s0 = arg0_1
    s1 = arg1_1
    s2 = arg2_1
    assert_size_stride(arg3_1, (s0, s1, s2), (s1*s2, s2, 1))
    # Topologically Sorted Source Nodes: [y], Original ATen: [mylib.foo]
    buf0 = torch.ops.mylib.foo.default(arg3_1)
    assert_size_stride(buf0, (u0, u1, u2), ((max(1, u1))*(max(1, u2)), max(1, u2), 1))
    u0 = buf0.size(0)
    u1 = buf0.size(1)
    u2 = buf0.size(2)
    del arg3_1
    if not u0 >= 0:
        raise RuntimeError('u0 >= 0')
    buf1 = None
    if not u1 >= 0:
        raise RuntimeError('u1 >= 0')
    buf2 = None
    if not u2 >= 0:
        raise RuntimeError('u2 >= 0')
    buf3 = None
    # Topologically Sorted Source Nodes: [z], Original ATen: [mylib.bar]
    buf4 = torch.ops.mylib.bar.default(buf0)
    assert_size_stride(buf4, (u0, u1, u2), ((max(1, u1))*(max(1, u2)), max(1, u2), 1))
    return (buf4, )
```

There is an assert `assert_size_stride(buf0, (u0, u1, u2), ((max(1, u1))*(max(1, u2)), max(1, u2), 1))`, which check for `u0`, `u1` and `u2` before defined or materialized these symbols, but I didn't find a proper API which could check for it, so I do this check here.

Reply:
I think you want:
https://github.com/pytorch/pytorch/blob/500ce29e4c1e46e0c16bc0b5281b842fb6c819ef/torch/fx/experimental/symbolic_shapes.py#L776

Other types of symbols should be fine.

Reply:
Thanks for your suggestion! Done.